UI/platform-osx.mm

```diff
@@ -238,6 +242,138 @@ - (void)drawRect:(NSRect)dirtyRect
+						}];
+				os_event_wait(block_finished);
+				os_event_destroy(block_finished);
+				audioStatus = [AVCaptureDevice
+					authorizationStatusForMediaType:
+						AVMediaTypeAudio];
+			}
+
+			permissionResponse = (MacPermissionStatus)audioStatus;
+		} else {
+			permissionResponse = kPermissionAuthorized;
+		}
+
+		blog(LOG_INFO, "[macOS] Permission for audio device access %s.",
+		     permissionResponse == kPermissionAuthorized ? "granted"
+								 : "denied");
+
+		return permissionResponse;
+		break;
+	}
+	case kVideoDeviceAccess: {
+		if (@available(macOS 10.14, *)) {
+			AVAuthorizationStatus videoStatus = [AVCaptureDevice
+				authorizationStatusForMediaType:AVMediaTypeVideo];
+
+			if (videoStatus == AVAuthorizationStatusNotDetermined &&
+			    prompt_for_permission) {
+				os_event_t *block_finished;
+				os_event_init(&block_finished,
+					      OS_EVENT_TYPE_MANUAL);
+				[AVCaptureDevice
+					requestAccessForMediaType:AVMediaTypeVideo
+						completionHandler:^(
+							BOOL granted
+							__attribute((unused))) {
+							os_event_signal(
+								block_finished);
+						}];
+
+				os_event_wait(block_finished);
+				os_event_destroy(block_finished);
+				videoStatus = [AVCaptureDevice
+					authorizationStatusForMediaType:
+						AVMediaTypeVideo];
+			}
+
+			permissionResponse = (MacPermissionStatus)videoStatus;
+		} else {
+			permissionResponse = kPermissionAuthorized;
+		}
+
+		blog(LOG_INFO, "[macOS] Permission for video device access %s.",
+		     permissionResponse == kPermissionAuthorized ? "granted"
+								 : "denied");
+
+		return permissionResponse;
+		break;
+	}
+	case kScreenCapture: {
+		if (@available(macOS 10.15, *)) {
+			permissionResponse = (CGPreflightScreenCaptureAccess()
+						      ? kPermissionAuthorized
+						      : kPermissionDenied);
+
+			if (permissionResponse != kPermissionAuthorized &&
+			    prompt_for_permission) {
+				permissionResponse =
+					(CGRequestScreenCaptureAccess()
+						 ? kPermissionAuthorized
+						 : kPermissionDenied);
+			}
+
+		} else {
+			permissionResponse = kPermissionAuthorized;
+		}
+
+		blog(LOG_INFO, "[macOS] Permission for screen capture %s.",
+		     permissionResponse == kPermissionAuthorized ? "granted"
+								 : "denied");
+
+		return permissionResponse;
+		break;
+	}
+	case kAccessibility: {
+		permissionResponse = (AXIsProcessTrusted()
+					      ? kPermissionAuthorized
+					      : kPermissionDenied);
+
+		if (permissionResponse != kPermissionAuthorized &&
+		    prompt_for_permission) {
+			NSDictionary *options = @{
+				(__bridge id)kAXTrustedCheckOptionPrompt: @YES
+			};
+			permissionResponse = (AXIsProcessTrustedWithOptions(
+						      (CFDictionaryRef)options)
+						      ? kPermissionAuthorized
+						      : kPermissionDenied);
+		}
+
+		blog(LOG_INFO, "[macOS] Permission for accessibility %s.",
+		     permissionResponse == kPermissionAuthorized ? "granted"
+								 : "denied");
+		return permissionResponse;
+		break;
```

Comment:
Is this standard in Objective C and/or consistent with our other uses of switch statements?

Reply:
Removed the early returns, as every block assigns a correct value to `permissionresponse`.